examples/groupchat.c

```diff
@@ -0,0 +1,409 @@
+	return true;
+}
+
+static void channel_poll(meshlink_handle_t *mesh, meshlink_channel_t *channel, size_t len) {
+	(void)len;
+
+	fprintf(stderr, "Channel to '%s' connected\n", channel->node->name);
+	meshlink_set_channel_poll_cb(mesh, channel, NULL);
+}
+
+static void node_status(meshlink_handle_t *mesh, meshlink_node_t *node, bool reachable) {
+	(void)mesh;
+
+	if(reachable) {
+		fprintf(stderr, "%s joined.\n", node->name);
+	} else {
+		fprintf(stderr, "%s left.\n", node->name);
+	}
+}
+
+static meshlink_node_t **nodes;
+static size_t nnodes;
+
+static void parse_command(meshlink_handle_t *mesh, char *buf) {
+	char *arg = strchr(buf, ' ');
+	char *arg1 = NULL;
+	meshlink_submesh_t *s = NULL;
+	size_t i;
+
+	if(arg) {
+		*arg++ = 0;
+
+		arg1 = strchr(arg, ' ');
+
+		if(arg1) {
+			*arg1++ = 0;
+		}
+
+	}
+
+	if(!strcasecmp(buf, "invite")) {
+		char *invitation;
+
+		if(!arg) {
+			fprintf(stderr, "/invite requires an argument!\n");
+			return;
+		}
+
+		if(arg1) {
+			static meshlink_submesh_t **submesh = NULL;
+			static size_t nmemb = 0;
+
+			submesh = devtool_get_all_submeshes(mesh, submesh, &nmemb);
+
+			if ( !submesh || !nmemb ) {
+				fprintf(stderr, "Group does not exist!\n");
+				return;
+			}
+
+			for (i = 0; i < nmemb; i++) {
+				if ( !strcmp( arg1, submesh[i]->name) ) {
+					s = submesh[i];
+					break;
+				} else {
+					s = NULL;
+				}
+			}
+
+			if ( !s ) {
+				fprintf(stderr, "Group is not yet creted!\n");
+				return;
+			}
+		}
+
+		invitation = meshlink_invite(mesh, s, arg);
+
+		if(!invitation) {
+			fprintf(stderr, "Could not invite '%s': %s\n", arg, meshlink_strerror(meshlink_errno));
+			return;
+		}
+
+		fprintf(stderr, "Invitation for %s: %s\n", arg, invitation);
+		free(invitation);
+	} else if(!strcasecmp(buf, "canonical")) {
+		bool set;
+		char *host = NULL, *port = NULL;
+
+		if(!arg) {
+			fprintf(stderr, "/canonical requires an argument!\n");
+			return;
+		}
+
+		if ( (0 == strncasecmp(arg, "-h", 2)) && (strlen(arg) > 2) ) {
+			host = arg + 2;
+		} else if ( (0 == strncasecmp(arg, "-p", 2)) && (strlen(arg) > 2) ) {
+			port = arg + 2;
+		} else {
+			fprintf(stderr, "Unknown argument: %s!\n", arg);
+			return;
+		}
+
+		if ( arg1 ) {
+			if ( (0 == strncasecmp(arg1, "-h", 2)) && (strlen(arg1) > 2) ) {
+				host = arg1 + 2;
+			} else if ( (0 == strncasecmp(arg1, "-p", 2)) && (strlen(arg1) > 2) ) {
+				port = arg1 + 2;
+			} else {
+				fprintf(stderr, "Unknown argument: %s!\n", arg1);
+				return;
+			}
+		}
+
+		if ( !host && !port ) {
+			fprintf(stderr, "Unable to set Canonical address because no valid arguments are found!\n");
+			return;
+		}
+
+		set = meshlink_set_canonical_address(mesh, meshlink_get_self(mesh), host, port);
+
+		if(!set) {
+			fprintf(stderr, "Could not set canonical address '%s:%s': %s\n", host, port, meshlink_strerror(meshlink_errno));
+			return;
+		}
+
+		fprintf(stderr, "Canonical address set as '%s:%s'\n", host, port);
+	} else if(!strcasecmp(buf, "group")) {
```

Comment:
The help text says the command is `/submesh`. You should either call it `group` everywhere, or `submesh` everywhere.

Reply:
I have missed it. But now I have changed it to /group